crates/sui-core/src/authority/authority_store.rs

```diff
@@ -62,52 +69,95 @@ pub struct SuiDataStore<const ALL_OBJ_VER: bool, const USE_LOCKS: bool, S> {
     /// composite key of the SuiAddress of their owner and the object ID of the object.
     /// This composite index allows an efficient iterator to list all objected currently owned
     /// by a specific user, and their object reference.
+    ///
+    /// Consistency requirements:
+    /// - Only exists to satisfy read queries from clients, and should be moved to IndexStore
+    /// - Eventual consistency with respect to certificate processing - the owner index must
+    ///   eventually be updated after a certificate is processed.
+    /// - Read-after-write consistency can be provided if required, such that if client has a
+    ///   signed effects cert they cannot be served the old owner_index entry.
     owner_index: DBMap<(SuiAddress, ObjectID), ObjectRef>,
 
     /// This is map between the transaction digest and transactions found in the `transaction_lock`.
     /// NOTE: after a lock is deleted (after a certificate is processed) the corresponding entry here
     /// could be deleted, but right now this is only done on gateways, not done on authorities.
+    ///
+    /// Consistency requirements:
+    /// - Must never contain an entry for a TransactionDigest unless locks have been succesfully
+    ///   acquired for that tx.
+    /// - Read-after-write consistency for the client: If the Validator::transaction RPC returns
+    ///   successfully, this map must be up to date. If the RPC fails before returning, the client
+    ///   retries, which will eventually cause this map to be up to date.
     transactions: DBMap<TransactionDigest, TransactionEnvelope<S>>,
 
     /// This is a map between the transaction digest and the corresponding certificate for all
     /// certificates that have been successfully processed by this authority. This set of certificates
     /// along with the genesis allows the reconstruction of all other state, and a full sync to this
     /// authority.
+    ///
+    /// Consistency requirements:
+    /// - Read-after-write with respect to Validator::confirmation_transaction
+    /// - A certificate must be added to this map *before* the certificate's input object
+    ///   locks are deleted and the output object locks are initialized
     certificates: DBMap<TransactionDigest, CertifiedTransaction>,
 
     /// The map between the object ref of objects processed at all versions and the transaction
     /// digest of the certificate that lead to the creation of this version of the object.
     ///
     /// When an object is deleted we include an entry into this table for its next version and
     /// a digest of ObjectDigest::deleted(), along with a link to the transaction that deleted it.
+    ///
+    /// Consistency requirements:
+    /// - Read-after-write with respect to Validator::confirmation_transaction
```

Comment:
Today we are mostly using this table as tombstone store: to know whether an object has been deleted/wrapped.

Reply:
@lxfind on that last note: I've been rethinking our decision to keep all objects and selectively GC later, because it complicates the read path.  It might be better to use a separate table to explicitly keep stuff that needs to be kept.  But sorry that's a different conversation.

Regarding knowing if something has been wrapped: could that be solved different ways, such as your idea to keep a separate metadata table for objects?

Reply:
I think the small complication to the read path more than pays for itself by simplifying my proposal here: https://www.notion.so/mystenlabs/Safe-TX-storage-and-sequencing-83e94806fac34910b867b5b9352fe698

If parent_sync is truly no longer necessary, we could use a simpler tombstone design, but I strongly feel we shouldn't go back to deleting entries from the objects table.

Reply:
Things have to be deleted sooner or later, it's just a matter of when it's safe.  I think there are multiple options here.